Test

mororor opened the pull request:
Added tests to hyperlib and also did some changes to library and archive in the process

library: 
 - added a _properArchiveTypes() function to prevent false loading of archiveTypes

archive:
 - added a question (in comments) to _ready()
 - changed isAuthorized() to return a promise of a db.authorized()-call
 - changed setShare() to also stop sharing if share is set to false
 - added stopShare() function

tests:
 - most of them are finished and perform pretty well
 - in the archive-sharing-test is a workaround included to prevent the leaking of network handles (which would prevent the test from ending). The leaking and the workaround is not fully understood. The multiwriter-rewrite of discovery-swarm hopefully solves that.
 - the archive mounting test still throws a lot of errors. Some of the behavior seems unintended. Maybe refactoring the mounting stuff. I could also have a look into that.

mororor committed:
Implemented some basic tests on library.js and archive.js and added stopShare() to archive.js

Adding stopShare() to archive.js was necessary to allow tape to
terminate. Anyhow, closing connections seems to be a logical behaviour
after setting status { share: false }. Still needs implementation.

mororor committed:
Added event emitters to network actions

This was primarily needed for testing purposes but its probabily useful to
emit something in case of network events later on.

mororor committed:
Some more tests on library and coverage report

The tests now cover 100% of library.js, I also included the coverage report.

mororor committed:
in index.js

there is a discrancy between letting the tests work and letting the archipel-core work.
I put to the core-working version.

mororor committed:
tydied up

mororor committed:
some more tests, library.js: added properArchiveTypes()

mororor committed:
added coverage/ to .gitignore & added coverage script to package.json

mororor committed:
Changes in archive.js

 - Changed isAuthorized() to get the current state from db.authorized 
and return it as new Promise.
 - Added a question as comment to ready()

mororor committed:
Finished tests:

- got coverage to relative high level in terms of lines
- archive synchronizations tests run now
- archive mounts test still throws a lot of error. There is some 
inconsistant behaviour, at least as far as I can tell. refactoring?

mororor committed:
coverage report

mororor committed:
changed .gitignore

mororor committed:
removed coverage, to get it out of the git repo

Frando commented:
looks good! added a couple small comments. tests look great. will look at these in more detail later, but i might also just merge later today. I have some changes to hyperlib in the pipeline, so having tests in place by then is best I think.

Frando committed:
Fix dependencies.

Frando committed:
Streamline ready function.

Frando committed:
Normalize emitted event names, fix bug in swarm close.

Frando committed:
Improve sharing test.

mororor committed:
Worked on tests

Replaced some tape.deepEquals(foo.getState(), {...}) by more explicit
tape.equal(foo.getState().bar, ...)
Worked a little on the "archive: mounts" test. three of the tests are
expected to fail, since functionality seems not to be implemented as
intended.

Frando commented:
Merged!
